### PR TITLE
Display errors from Ancestor entities

### PR DESCRIPTION
**Pull request by benjdlambert**
Signed-off-by: blam <ben@blam.sh>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jhaals** [comment]
Screenshot 2021-10-04 at 17 36 55
Screenshot 2021-10-04 at 17 36 48

**Comment by benjdlambert** [comment]
Got some failing tests here @jhaals~

**Comment by freben** [comment]
Closes #2876

File: plugins/catalog-backend/src/next/NextEntitiesCatalog.ts [hunk 1]
```diff
@@ -189,7 +189,6 @@ export class NextEntitiesCatalog implements EntitiesCatalog {
       current = todo.pop()
     ) {
       const currentRef = stringifyEntityRef(current);
-      seenEntityRefs.add(currentRef);
```
[the remark this answers is not in the document]
Comment (jhaals) [reply plugins/catalog-backend/src/next/NextEntitiesCatalog.ts]: removed it for now, please follow up with this in a separate PR as this is already big enough 😅

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-backend/src/next/NextEntitiesCatalog.ts]: sounds good!

File: packages/catalog-client/src/CatalogClient.ts [hunk 2]
```diff
@@ -44,6 +47,18 @@ export class CatalogClient implements CatalogApi {
...
   }
 
+  async getEntityAncestors(
+    request: CatalogEntityAncestorsRequest,
+    options?: CatalogRequestOptions,
+  ): Promise<CatalogEntityAncestorsResponse> {
+    const { kind, namespace, name } = parseEntityRef(request.entityRef);
+    return await this.requestRequired(
+      'GET',
+      `/entities/by-name/${kind}/${namespace}/${name}/ancestry`,
```
Comment (freben) [inline packages/catalog-client/src/CatalogClient.ts:57]: still need to `encodeURIComponent` each of these three

File: plugins/catalog/api-report.md [hunk 3]
```diff
@@ -382,7 +397,12 @@ export const FilteredEntityLayout: ({
 // Warning: (ae-missing-release-tag) "hasCatalogProcessingErrors" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
 //
 // @public (undocumented)
-export const hasCatalogProcessingErrors: (entity: Entity) => boolean;
+export const hasCatalogProcessingErrors: (
+  entity: Entity,
+  context: {
+    apis: ApiHolder;
+  },
+) => Promise<boolean>;
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog/api-report.md:405]: still no mention of this in the changeset though

[the remark this answers is not in the document]
Comment (benjdlambert) [reply plugins/catalog/api-report.md:405]: @jhaals shall we mention this?

**Review by freben — APPROVED** [review]
Unblocking approve, two nits tho

File: plugins/catalog/api-report.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhaals) [reply plugins/catalog/api-report.md:405]: Good point, I updated the changeset. @benjdlambert PTAL

**Pull request merged**